torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1292,6 +1292,31 @@ int computeDeltaMS(
       .count();
 }
 
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutErrorMsg(
+    const std::string& extraMsg) {
+  return c10::str(
+      logPrefix(),
+      "Received a dump signal due to a collective timeout from ",
+      extraMsg,
+      " and we will try our best to dump the debug info. ",
+      "Last enqueued NCCL work: ",
+      pgStatus_->lastEnqueuedSeq,
+      ", last completed NCCL work: ",
+      pgStatus_->lastCompletedSeq,
+      ".",
+      "This is most likely caused by incorrect usages of collectives, e.g., wrong ",
+      "sizes used across ranks, the order of collectives is not same for all ranks ",
+      "or the scheduled collective, for some reason, didn't run. Additionally, ",
+      "this can be caused by GIL deadlock or other reasons such as network errors or ",
+      "bugs in the communications library (e.g. NCCL), etc. ");
+}
+
+std::string ProcessGroupNCCL::getNCCLWatchdogTimeoutExitMsg() {
+  return c10::str(
+      logPrefix(),
+      "Exit processGroupNCCL after attempting to dump debug info, due to collective timeout or exception.");
```

Comment:
"Exit processGroupNCCL" doesn't sound right still.  (grammatically, and, exit isn't quite a precise term, it could mean 'returning'

you really need "terminating the process" in there somewhere to make it clear what is happening.  I think its ok to drop the pgnccl part since that shows up in the log line (filename/lineno is there) but if you want to put it you can put it like "ProcessGroupNCCL terminating the process after..."

Reply:
After a second thought, I think I shouldn't use "processGroupNCCL" at all. Let me change it.